### PR TITLE
Bug 1146744 - Don't call pair() on an already paired BT device

### PR DESCRIPTION
**Pull request by apuder**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=44f1cb4f1accf33ab78eeab93d9e671f87180d45

File: apps/system/js/nfc_handover_manager.js [hunk 1]
```diff
@@ -294,31 +294,94 @@ var NfcHandoverManager = {
...
+
+    var req = this.defaultAdapter.getPairedDevices();
+    req.onsuccess = () => {
+      var devices = req.result;
+      this._debug('# devices: ' + devices.length);
+      for (var i = 0; i < devices.length; i++) {
+        var device = devices[i];
+        this._debug('Address: ' + device.address);
+        this._debug('Connected: ' + device.connected);
+        if (device.address.toLowerCase() == mac.toLowerCase()) {
```
Comment (alivedise) [inline apps/system/js/nfc_handover_manager.js]: ===

File: apps/system/js/nfc_handover_manager.js [hunk 2]
```diff
@@ -294,31 +294,94 @@ var NfcHandoverManager = {
...
-   * @param {string} mac MAC address of the peer device
+   * Look for a paired device and invoke the appropriate callback function.
+   * @param {string} mac MAC address of the device
+   * @param {function} foundCb Found callback
+   * @param {function} notFoundCb Not found callback
    * @memberof NfcHandoverManager.prototype
    */
-  _doPairing: function _doPairing(mac) {
-    this._debug('doPairing: ' + mac);
+  _findPairedDevice: function _findPairedDevice(mac, foundCb, notFoundCb) {
```
Comment (alivedise) [inline apps/system/js/nfc_handover_manager.js]: nit: it's better to use promise pattern in the future.
usage => this._findPairedDevice(mac).then(function() {}).catch(function() {});

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=272ef5b1cdb46220e3be5ba99eebb35b1d0f80d6

**Pull request merged**